plugins/com.google.cloud.tools.eclipse.dataflow.core.test/src/com/google/cloud/tools/eclipse/dataflow/core/launcher/MajorVersionPipelineRunnerConfigurationTest.java

```diff
@@ -66,4 +72,21 @@ public void testDefaultRunnerInConfiguredRunners() {
     Set<PipelineRunner> runners = PipelineRunner.inMajorVersion(majorVersion);
     assertThat(runners, CoreMatchers.hasItem(defaultRunner));
   }
+
+  /** Ensure the default runner chosen is appropriate for the majorVersion. */
+  @Test
+  public void testFromLaunchConfiguration_defaultRunner() throws CoreException {
+    ILaunchConfiguration empty = mock(ILaunchConfiguration.class, new Answer() {
+      @Override
+      public Object answer(InvocationOnMock invocation) throws Throwable {
+        // for getAttribute(attr, default)
+        Object[] arguments = invocation.getArguments();
+        return arguments.length == 2 ? arguments[1] : null;
```

Comment:
See if this works: `when(empty.getAttribute(anyObject(), anyObject())).thenAnswer(AdditionalAnswers.returnsSecondArg());`

Reply:
Ooo, that's much nicer.